tests/HUBViewModelLoaderTests.m

```diff
@@ -674,6 +674,254 @@ - (void)testNilReloadPolicyAlwaysResultingInReload
+    
+    [self.loader loadViewModel];
+    
+    NSArray<id<HUBComponentModel>> * const componentModelsA = self.viewModelFromSuccessDelegateMethod.bodyComponentModels;
+    XCTAssertEqual(componentModelsA.count, 2u);
+    XCTAssertEqualObjects(componentModelsA[0].title, @"First component A");
+    XCTAssertEqualObjects(componentModelsA[1].title, @"First component B");
+    
+    [self.loader loadNextPageForCurrentViewModel];
+    
+    NSArray<id<HUBComponentModel>> * const componentModelsB = self.viewModelFromSuccessDelegateMethod.bodyComponentModels;
+    XCTAssertEqual(componentModelsB.count, 3u);
+    XCTAssertEqualObjects(componentModelsB[0].title, @"First component A");
+    XCTAssertEqualObjects(componentModelsB[1].title, @"First component B");
+    XCTAssertEqualObjects(componentModelsB[2].title, @"Second component A");
+}
+
+- (void)testPageIndexIncrementedForEachPaginatedLoadingChain
+{
+    HUBContentOperationMock * const contentOperation = [HUBContentOperationMock new];
+    
+    __block NSUInteger pageIndex = 0;
+    
+    contentOperation.paginatedContentLoadingBlock = ^(id<HUBViewModelBuilder> builder, NSUInteger newPageIndex) {
+        pageIndex = newPageIndex;
+        return YES;
+    };
+    
+    [self createLoaderWithContentOperations:@[contentOperation]
+                          connectivityState:HUBConnectivityStateOnline
+                           initialViewModel:nil];
+    
+    [self.loader loadViewModel];
+    
+    [self.loader loadNextPageForCurrentViewModel];
+    XCTAssertEqual(pageIndex, 1u);
+    
+    [self.loader loadNextPageForCurrentViewModel];
+    XCTAssertEqual(pageIndex, 2u);
+    
+    [self.loader loadNextPageForCurrentViewModel];
+    XCTAssertEqual(pageIndex, 3u);
+}
+
+- (void)testLoadingPaginatedContentMultipleTimesAppendsToQueue
+{
+    HUBContentOperationMock * const contentOperation = [HUBContentOperationMock new];
+    
+    contentOperation.paginatedContentLoadingBlock = ^(id<HUBViewModelBuilder> builder, NSUInteger pageIndex) {
+        NSString * const componentIdentifier = [NSString stringWithFormat:@"%@", @(pageIndex)];
+        [builder builderForBodyComponentModelWithIdentifier:componentIdentifier].title = @"Component";
+        return NO;
+    };
+    
+    [self createLoaderWithContentOperations:@[contentOperation]
+                          connectivityState:HUBConnectivityStateOnline
+                           initialViewModel:nil];
+    
+    [self.loader loadViewModel];
+    [self.loader loadNextPageForCurrentViewModel];
+    [self.loader loadNextPageForCurrentViewModel];
+    [self.loader loadNextPageForCurrentViewModel];
+    
+    id<HUBContentOperationDelegate> const contentOperationDelegate = contentOperation.delegate;
+    [contentOperationDelegate contentOperationDidFinish:contentOperation];
+    [contentOperationDelegate contentOperationDidFinish:contentOperation];
+    
+    // Make sure that we don't have any component models until the last operation is finished
```

Comment:
Well, we trust the API user. If they tell us to load multiple pages at once - we will. If not, they should load a new page, let Hubs render it - and then load another one. We want appending content using pagination to work in the same way as the normal content loading chain - that is - we only render once the queue is empty.